torch/csrc/distributed/Module.cpp

```diff
@@ -338,6 +367,271 @@ PyObject* THDPModule_recv(PyObject *_unused, PyObject *args)
   END_HANDLE_TH_ERRORS
 }
 
+
+PyObject* THDPModule_allReduceMultiGPU(PyObject *_unused, PyObject *args)
+{
+  HANDLE_TH_ERRORS
+  std::vector<at::Tensor> descriptors;
+  Py_ssize_t tmp_length;
+  std::size_t length;
+  THDGroup group;
+  THDReduceOp op;
+  PyObject* sequence;
+
+  if (PyTuple_GET_SIZE(args) != 3) {
+    goto invalid_arguments;
+  }
+
+  if (!PySequence_Check(PyTuple_GET_ITEM(args, 0))) {
+    goto invalid_arguments;
+  }
+
+  sequence = PySequence_Fast(PyTuple_GET_ITEM(args, 0), nullptr);
```

Comment:
This returns a new reference, so you will leak memory if you don't decref it. That's why I suggested using `THPObjectPtr`, because it will do it for you automatically (even in case of any errors).

Reply:
This is now fixed in the new commit